### ISSUE TITLE
RStudio with traefik

### ISSUE THREAD

**Issue by wahu33** [rocker-org/rocker#396]
Tried to install RStudio with traefik with following docker-compose.yml
`    version: "3.3"

    services:
      rstudio:
        image: rocker/rstudio:latest
        container_name: rstudio
        restart: unless-stopped

        volumes:
          - /etc/localtime:/etc/localtime:ro
          - /var/docker/rstudio:/home/rstudio
        environment:
          - PASSWORD=xxx
        labels:
          - "traefik.enable=true"
          - "traefik.http.routers.rstudio.entrypoints=http"
          - "traefik.http.routers.rstudio.rule=Host(`rstudio.example.com`)"
          - "traefik.http.services.rstudio.loadbalancer.server.port=8787"   
          - "traefik.docker.network=proxy"
        networks:
          - default
          - proxy
          
    networks:
      proxy:
        external: true
`
![Screenshot_2020-04-29 RStudio](https://user-images.githubusercontent.com/19956861/80583649-c9532280-8a10-11ea-972d-1545375fe9fc.png)

This brings a Alert-box "RStudio initialization Error" with text "Unable to connect to service".

**Comment by cboettig** [comment]
Apologies, but I think we need more context here.  Not familiar with `traefik` but I don't see why those labels would have any effect in this container?  If you're trying to configure a reverse proxy, there's plenty of nice examples out there with Caddy or NGINX, but usually you need the reverse proxy service to be running in its own container.  Maybe consider posting a minimal traefik example that does work (i.e. with a non-rocker container)?

**Comment by rlaiola** [comment]
@cboettig, the instructions on how to reproduce it can be found below.

1. Create a file as follows:
_docker-compose.yml_
    ````
    version: "3.3"

    services:

      traefik:
        image: "traefik:v2.6"
        container_name: "traefik"
        command:
          #- "--log.level=DEBUG"
          - "--api.insecure=true"
          - "--providers.docker=true"
          - "--providers.docker.exposedbydefault=false"
          - "--entrypoints.web.address=:80"
        ports:
          - "80:80"
          - "8080:8080"
        volumes:
          - "/var/run/docker.sock:/var/run/docker.sock:ro"

      rstudio:
        image: "rocker/rstudio:latest"
        container_name: "rstudio"
        restart: "unless-stopped"
        volumes:
          - "/etc/localtime:/etc/localtime:ro"
          - "/var/docker/rstudio:/home/rstudio"
        environment:
          - PASSWORD=xxx
        labels:
          - "traefik.enable=true"
          - "traefik.http.routers.rstudio.rule=Host(`rstudio.localhost`)"
          - "traefik.http.routers.rstudio.entrypoints=web"
          - "traefik.http.services.rstudio.loadbalancer.server.port=8787"
    ````
1. Open a Terminal window (e.g., Bash), set the work folder to the same place where you saved the file _`docker-compose.yml`_ and run the following command:
`docker-compose up -d`
1. Open a Web browser window, visit the URL http://rstudio.localhost and log in (Username: rstudio / Password: xxx). After a while, the error dialog appears.
    <img width="689" alt="Screen Shot 2022-03-15 at 09 02 48" src="https://user-images.githubusercontent.com/4059310/158377413-5a4cf2f8-2c16-4abf-9e8a-59e0124b6534.png">
1. To stop it:
`docker-compose down`

**Bottom line: The problem surely occurs if the user that calls `docker-compose up -d` does not have privilleges to create the folder _`/var/docker/rstudio`_ on the host machine (see below). I do not see it as an issue and if @wahu33 has nothing to add, I'd recommend closing it.**
<img width="689" alt="Screen Shot 2022-03-15 at 09 36 51" src="https://user-images.githubusercontent.com/4059310/158381950-bb82851a-0404-4efd-96df-c8f7bbc17a1e.png">

Cheers!

**Comment by eitsupi** [comment]
Thanks @rlaiola.
I'll close this as it seems to be a problem caused by an improper bindmount (I don't think it's proper to bindmount whole of a user's home directories on a Docker image in general).
